types/d3-array/d3-array-tests.ts

```diff
@@ -380,11 +567,17 @@ numbersArray = d3Array.range(1, 10, 0.5);
 // shuffle() -------------------------------------------------------------------
 
 mergedArray = d3Array.shuffle(mergedArray);
-
 mergedArray = d3Array.shuffle(mergedArray, 1);
-
 mergedArray = d3Array.shuffle(mergedArray, 1, 3);
 
+mergedArray = d3Array.shuffle(readonlyMergedArray);
```

Comment:
I'm not sure how I feel about maintaining that many overloads. The heart of the issue is that TypeScript doesn't consider the readonly-ness of a property when computing assignability of function arguments.

If `shuffle` is the only function that does in-place modification, maybe it's worth relaxing this to `ArrayLike<T>` as well and just noting in the documentation that it mutates its argument?